lib/AST/InlinableText.cpp

```diff
@@ -189,41 +189,78 @@ struct ExtractInactiveRanges : public ASTWalker {
 };
 } // end anonymous namespace
 
+static void appendRange(
+  SourceManager &sourceMgr, SourceLoc start, SourceLoc end,
+  SmallVectorImpl<char> &scratch) {
+  unsigned bufferID = sourceMgr.findBufferContainingLoc(start);
+  unsigned offset = sourceMgr.getLocOffsetInBuffer(start, bufferID);
+  unsigned endOffset = sourceMgr.getLocOffsetInBuffer(end, bufferID);
+
+  // Strip comments from the chunk before adding it by re-lexing the range.
+  LangOptions FakeLangOpts;
+  Lexer lexer(FakeLangOpts, sourceMgr, bufferID, nullptr, LexerMode::Swift,
+    HashbangMode::Disallowed, CommentRetentionMode::ReturnAsTokens,
+    offset, endOffset);
+
+  SmallVector<SourceRange, 8> nonCommentRanges;
+  SourceLoc nonCommentStart = start;
+  Token token;
+
+  // Re-lex the range, and skip the full text of `tok::comment` tokens.
+  while (true) {
+    lexer.lex(token);
+
+    if (token.is(tok::eof))
+      break;
+
+    if (token.is(tok::comment)) {
+      // Grab the range from the last non-comment token to the beginning of this comment
+      // token.
+      SourceLoc commentLoc = token.getLoc();
+
+      SourceRange range { nonCommentStart, commentLoc };
+      nonCommentRanges.append(1, range);
```

Comment:
Any reason not to do 

```suggestion
      nonCommentRanges.push_back(range);
```

Same below